transaction/src/model/validated_transaction.rs

```diff
@@ -35,8 +43,25 @@ impl ExecutableTransaction for ValidatedTransaction {
         &self.instructions
     }
 
-    fn signer_public_keys(&self) -> &[PublicKey] {
-        &self.signer_public_keys
+    fn initial_proofs(&self) -> Vec<NonFungibleAddress> {
+        match &self.actor {
+            ValidatedTransactionActor::User(signer_public_keys) => signer_public_keys
+                .iter()
+                .map(|k| match k {
+                    PublicKey::EddsaEd25519(pk) => NonFungibleAddress::new(
+                        ED25519_TOKEN,
+                        NonFungibleId::from_bytes(pk.to_vec()),
+                    ),
+                    PublicKey::EcdsaSecp256k1(pk) => {
+                        NonFungibleAddress::new(ECDSA_TOKEN, NonFungibleId::from_bytes(pk.to_vec()))
+                    }
+                })
+                .collect(),
+            ValidatedTransactionActor::Supervisor => vec![NonFungibleAddress::new(
```

Comment:
It would be nice if there was a function to create this System Badge, as we're duplicating this code in a few places?